lighthouse-core/test/lib/stack-packs-test.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env jest */
+
+const lighthouseStackPacks = require('lighthouse-stack-packs');
+const defaultConfig = require('../../config/default-config.js');
+const Config = require('../../config/config.js');
+
+function getAuditIds() {
+  const config = new Config(defaultConfig);
+  return config.audits.map(a => a.implementation.meta.id);
+}
+
+describe('lighthouse-stack-packs', () => {
+  it('snapshot all strings', () => {
+    /* eslint-disable max-len */
+    expect(lighthouseStackPacks).toMatchInlineSnapshot(`
```

Comment:
Snapshot testing the entirety of the stack packs seems a little excessive? We don't have a snapshot test for all of `axe.min.js`, for instance :)

Reply:
The test that @patrickhulce suggested does a more narrow check for a problem we've seen already.

This one is larger, a catch all for whatever strings that end up in Lighthouse. A chance for us to stop something bad from being published.

Reply:
Maybe what we're looking for here then @connorjclark is a comment expressing what a reviewer should look for here? Right now I'm not sure what changes to this snapshot I should *expect* and what changes to this would be *bad* (which was the same core problem in missing the `-` -> `_` change in the sample_v2.json)

I think what @brendankenny was saying if we're trying to prevent something like "all of these strings disappeared" then we can test that with a narrower snapshot assertion than having to inspect all of these strings manually.

Reply:
>  inspect all of these strings manually.

But we should check these manually. The release process for this package, while still handled by Googlers, is not 100% in our direct control. It's maybe not a realistic attack vector, but a quick 5-10s glance every quarter when we update this strings is not an expensive thing.

Reply:
> But we should check these manually. The release process for this package, while still handled by Googlers, is not 100% in our direct control. It's maybe not a realistic attack vector, but a quick 5-10s glance every quarter when we update this package is not an expensive thing.

That's not really a test (unit or otherwise), though, it's a "hey human, check all of this for...something" (which we're also doing for the same strings in `en-US.json` and x50 for all the other languages when the translations come in).

If this is the chief need for this check, we should be explicit about it and do a periodic static copy of `lighthouse-stack-packs` into `third-party/`. Then it's crystal clear that what really needs to happen is a code review.

That seems like overkill, though. Really we care about things like, are joomla strings returned when `getStackPacks` is called with joomla detected, are strings displayed by the renderer when they're in `sample_v2` like they are, etc. Pushing the basic responsibilities to the separate repo seems like part of the point of having it in a separate repo.

Reply:
> is not 100% in our direct control. It's maybe not a realistic attack vector

and if we should be worried about this, the renderer is ok because no string gets privileged treatment, but we should really bulk up `collect-strings` and do some `unknown` checking because each string in there costs us $$ for translations :)

Reply:
updated the tests